spec/features/search/search_spec.rb

```diff
@@ -1,17 +1,38 @@
 require "rails_helper"
 
-describe "Search page", type: :feature do
+describe "Seach page", type: :feature do
 
-  it "shows all items when searching for nothing" do
-    @item = create(:item)
+  before do
     @item_book = create(:item_book)
-
+    @item_beamer = create(:item_beamer)
+    @item_whiteboard = create(:item_whiteboard)
     visit search_path
-    expect(page).to have_text(@item.name)
+  end
+
+  it "partial matching works for title" do
+    page.fill_in "search", with: "Ruby"
+    click_button("submit")
     expect(page).to have_text(@item_book.name)
+    expect(page).to have_link href: item_path(@item_book)
+    expect(page).not_to have_text(@item_beamer.name)
+    expect(page).not_to have_text(@item_whiteboard.name)
+  end
 
-    expect(page).to have_link href: item_path(@item)
+  it "partial matching works for description" do
+    page.fill_in "search", with: "book"
+    click_button("submit")
+    expect(page).to have_text(@item_book.name)
     expect(page).to have_link href: item_path(@item_book)
+    expect(page).not_to have_text(@item_beamer.name)
+    expect(page).not_to have_text(@item_whiteboard.name)
+  end
+
+  it "partial matching works for empty string" do
+    page.fill_in "search", with: ""
+    click_button("submit")
+    expect(page).not_to have_text(@item_book.name)
+    expect(page).not_to have_text(@item_beamer.name)
+    expect(page).not_to have_text(@item_whiteboard.name)
   end
 
```

Comment:
We should add a test that checks if the page displays all (or none) of the items when the search bar is empty

Reply:
Good idea! We would wait until we have an answer from Lucas :)

Reply:
It's done 🥳